Wrappers/Python/cil/processors/TransmissionAbsorptionConverter.py

```diff
@@ -84,47 +83,49 @@ def process(self, out=None):
         arr_in = data.as_array()
         arr_out = out.as_array()
 
-        #whitelevel
-        if self.white_level != 1:
-            numpy.divide(arr_in, self.white_level, out=arr_out)
-            arr_in = arr_out
-
-        #threshold
-        if self.min_intensity > 0:
-            numpy.clip(arr_in, self.min_intensity, None, out=arr_out)
-            arr_in = arr_out
-
-        #beer-lambert
+        # we choose an arbitrary chunk size of 6400, which is a multiple of 32, to allow for efficient threading
         chunk_size = 6400
         num_chunks = data.size // chunk_size
+        # Use numba if _accelerated is True and if the number of chunks is greater than 5, to avoid the overhead of threading when the data is small
         if (self._accelerated) & (num_chunks > 5):
             remainder = data.size % chunk_size
             num_threads_original = numba.get_num_threads()
             numba.set_num_threads(cil_mp.NUM_THREADS)
-            numba_loop(arr_in, num_chunks, chunk_size, remainder, arr_out)
-            # reset the number of threads to the original value
+            numba_loop(arr_in, num_chunks, chunk_size, remainder, 1/self.white_level, self.min_intensity,  arr_out)
             numba.set_num_threads(num_threads_original)
 
         else:
-            numpy.log(arr_in,out=arr_out)
+            if self.white_level != 1:
+                numpy.multiply(arr_in, 1/self.white_level, out=arr_out)
+                arr_in = arr_out
+
+            if self.min_intensity > 0:
+                numpy.clip(arr_in, self.min_intensity, None, out=arr_out)
+                arr_in = arr_out
+                
+            numpy.log(arr_in, out=arr_out)
             numpy.negative(arr_out,out=arr_out)
 
         out.fill(arr_out)
 
         return out
     
 @numba.njit(parallel=True)
-def numba_loop(arr_in, num_chunks, chunk_size, remainder, arr_out):
+def numba_loop(arr_in, num_chunks, chunk_size, remainder, multiplier, min_intensity, arr_out):
     in_flat = arr_in.ravel()
     out_flat = arr_out.ravel()
     for i in numba.prange(num_chunks):
         start = i * chunk_size
         end = start + chunk_size
-        out_flat[start:end] = -numpy.log(in_flat[start:end])
+        out_flat[start:end] = numpy.multiply(in_flat[start:end], multiplier)
```

Comment:
is it worth checking that this is not 1 before doing the multiply? (without numba we check this)